runcommandonset/tests/runcommandonset.set.tests.ps1

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'tests for runcommandonset set' {
+    BeforeAll {
+        $oldPath = $env:DSC_RESOURCE_PATH
+        $env:DSC_RESOURCE_PATH = Join-Path $PSScriptRoot ".."
+        $yaml = @"
+executable: pwsh
+arguments:
+- -Command
+- echo hello world
+"@
+    }
+
+    AfterEach {
+        if (Test-Path $TestDrive/output.txt) {
+            Remove-Item -Path $TestDrive/output.txt
+        }
+    }
+
+    AfterAll {
+        $env:DSC_RESOURCE_PATH = $oldPath
+    }
+
+    It 'Input can be sent to the resource via stdin json' {
+        $json = @"
+        {
+            "executable": "pwsh",
+            "arguments": ["-Command", "echo hello world"],
+        }
+"@
+
+        $json | dsc resource set -r Microsoft.DSC.Transitional/RunCommandOnSet > $TestDrive/output.txt
```

Comment:
You don't have to redirect the output to a file, you can always assign it to a variable: $output = $json | dsc ...